src/link/MachO.zig

```diff
@@ -660,11 +652,18 @@ pub fn resolveLibSystem(
             "libSystem",
         )) break :success;
 
-        const dir = try comp.zig_lib_directory.join(tmp_arena, &[_][]const u8{ "libc", "darwin" });
-        const lib_name = try std.fmt.allocPrint(tmp_arena, "libSystem.{d}", .{
-            self.base.options.target.os.version_range.semver.min.major,
-        });
-        if (try accessLibPath(tmp_arena, &test_path, &checked_paths, dir, lib_name)) break :success;
+        switch (self.base.options.libc_provider) {
+            .none => unreachable,
+            .installation => unreachable,
```

Comment:
Can these really not be reached? From your changes to `Compilation.zig` it looks like at least `.installation` can be reached if we fail to extract the sysroot path from the detected include dirs. If that's the case, we should skip resolving the path to libSystem and report undefined symbols errors, rather than panic/UB here.

Reply:
With the latest commit only unreachable case is `.none` and it should be impossible for this code site.